iamctl/pkg/claims/export.go

```diff
@@ -0,0 +1,115 @@
+/**
+* Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+*
+* WSO2 LLC. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package claims
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"mime"
+	"os"
+	"path/filepath"
+
+	"github.com/wso2-extensions/identity-tools-cli/iamctl/pkg/utils"
+)
+
+func ExportAll(exportFilePath string, format string) {
+
+	// Export all claim dialects with related claims.
+	log.Println("Exporting Claim Dialects...")
+	exportFilePath = filepath.Join(exportFilePath, utils.CLAIMS)
+
+	if _, err := os.Stat(exportFilePath); os.IsNotExist(err) {
+		os.MkdirAll(exportFilePath, 0700)
+	} else {
+		if utils.TOOL_CONFIGS.AllowDelete {
+			utils.RemoveDeletedLocalResources(exportFilePath, getDeployedClaimDialectNames())
+		}
+	}
+
+	claimdialects, err := getClaimDialectsList()
+	if err != nil {
+		log.Println("Error: when getting Claim Dialect IDs.", err)
+	} else {
+		// if !utils.AreSecretsExcluded(utils.TOOL_CONFIGS.ApplicationConfigs) {
+		// 	log.Println("Warn: Secrets exclusion cannot be disabled for claim dialects. All secrets will be masked.")
```

Comment:
Since we do not need or support secret exclusion for claims, IMO it's not necessary to handle situations where a user adds the EXCLUDE_SECRETS config.

Reply:
Removed all commented out code